### ISSUE TITLE
update HCA ontology to OLS 4

### ISSUE THREAD

**Issue by amnonkhen** [ebi-ait/dcp-ingest-central#935]
**overview**
upgrade OLS to 4 on HCAO.

** Tasks **

- [x] get information about new version and upgrade process from ontologies team
- [ ] upgrade locally and test
- [ ] test upgrade on dev

**Context**
The [gender schema PR](https://github.com/ebi-ait/hca-ebi-wrangler-central/issues/844) is blocked by an ontology issue since November 2022
From release 1.0.45 a new set of terms was loaded from NCIT, under "[Gender](https://ontology.staging.archive.data.humancellatlas.org/ontologies/hcao/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FNCIT_C17357)". However, our validators are not recognising any of the new terms because they are missing the "obo_id" field , which we use to query the service and verify terms are present. You can see an example api response for the imported terms [here](https://ontology.staging.archive.data.humancellatlas.org/api/ontologies/hcao/terms?iri=http://purl.obolibrary.org/obo/NCIT_C17357).

Zoe advised us to reached out to Henriette from the OLS team and ask for her help
From Henriette
>Unfortunately I cannot help you to resolve the missing obo_id. The version of OLS you are on is incredibly old and not supported. The obo_id is available on the main OLS (https://www.ebi.ac.uk/ols/api/ontologies/hcao/terms?iri=http://purl.obolibrary.org/obo/NCIT_C17357) and on the new OLS4 (https://www.ebi.ac.uk/ols4/api/ontologies/hcao/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FNCIT_C17357) for this term.
Some suggestions as to how you can deal with this:
>1. Use OLS4 instead of your instance of OLS. This will be the recommended approach.
>2. Update the code to use either use the obo_id or the short_form (which is available based on API response)
>3. Upgrade the OLS version of your instance. This is a point we have raised a number of times with HCA in the past and the problem is that no one seems to know where this server is in order to ssh into it. Another option is to setup a completely new server with a new instance.

**Actions**
- [ ] discuss priority of task
- [x] chose one of the suggested approaches

**Comment by Wkt8** [comment]
Amnon to go with path 1 (updating to OLS4) in dev.

**Comment by amnonkhen** [comment]
Send the ontology team a few questions about the upgrade process on the [#hca-ontologies slack channel](https://spot-ebi.slack.com/archives/C03L348KU4X).

**Comment by serjoshua** [comment]
Good day, Josh here from the OLS team. Thanks to Bradley, we were directed to this ticket regarding an inquiry upgrading to OLS4. I saw that Henriette already responded to a separate inquiry as well regarding an obo_id, which I concur.

Upon checking the Dockerfile you shared, there will definitely be some updates that need to be done if you are going to run OLS4 locally or on your own server. Luckily, we believe, it is easier now than ever. You can check [OLS4 GitHub](https://github.com/EBISPOT/ols4) page regarding this matter. James made a detailed explanation how the new parts of OLS4 were implemented and how to run them using Docker. As a summary, there are four major Docker containers (excluding `frontend`) that will be of interest to you: `dataload`, `solr`, `neo4j` and `backend`. The `dataload` turns ontologies to datasets, loads them to `solr` and `neo4j` and then starts both containers on ports 8983 and 7474, respectively. The `backend` hosts the API above the `solr` and `neo4j` instances. To use your own ontology, you have to provide a configuration in JSON format to `OLS4_CONFIG` environment variable. Sample files can be found inside the `dataload/configs` directory of the project. Again, more detailed instructions and information are available in our GitHub page.

If you are using OLS (Production) APIs, you could easily change the URL paths from ols to ols4. OLS4 APIs are meant to be backward compatible. Any discrepancies will be considered a bug. Feel free to raise an issue in our GitHub page for any concerns regarding the use of APIs. Thanks.

**Comment by amnonkhen** [comment]
A discussion with Enrique, James and Amnon held on 10/7/2023.
This will be handled in a new ticket #963
